src/many-types/src/ledger.rs

```diff
@@ -1,17 +1,36 @@
-use crate::Percent;
+use crate::{cbor::CborNull, cbor_type_decl, Either, Percent};
+use many_error::ManyError;
 use many_identity::Address;
 use minicbor::data::{Tag, Type};
 use minicbor::{encode, Decode, Decoder, Encode, Encoder};
 use num_bigint::{BigInt, BigUint};
 use num_traits::Num;
 use serde::de::Unexpected;
 use serde::Deserialize;
+use std::collections::BTreeMap;
 use std::fmt::{Display, Formatter};
 use std::ops::Shr;
+use std::str::FromStr;
 
 /// A Symbol is represented by a non-anonymous identity.
 pub type Symbol = Address;
 
+/// A map of owners => tokens.
+pub type LedgerTokensAddressMap = BTreeMap<Address, TokenAmount>;
+
+pub type TokenMaybeOwner = Either<Address, CborNull>;
+
+impl FromStr for Either<Address, CborNull> {
```

Comment:
It's only used in the `ledger` CLI, so we can parse it with `clap`.

See https://github.com/liftedinit/many-framework/pull/293/files#diff-09f68f1b2106d3dfae9f7eefa15f0db9d8614fbe96da917124c8e2be0dfea891R66

I don't think it will affect anything else (for the moment) since I created `CborNull` for the Token feature.

Reply:
If that's your only usage I'd rather use Clap's `value_parser` (documentation https://docs.rs/clap/latest/clap/_derive/_tutorial/index.html#validated-values).